upup/pkg/fi/cloudup/gce/instancegroups.go

```diff
@@ -257,3 +263,30 @@ func matchInstanceGroup(mig *compute.InstanceGroupManager, c *kops.Cluster, inst
 	}
 	return matches[0], nil
 }
+
+func addCloudInstanceData(cm *cloudinstances.CloudInstance, instance *compute.Instance) {
+	cm.MachineType = LastComponent(instance.MachineType)
+	cm.Status = instance.Status
+	if instance.Status == "RUNNING" {
+		cm.State = cloudinstances.CloudInstanceStatusUpToDate
+	}
+	isControlPlane := false
+	for k := range instance.Labels {
+		if !strings.HasPrefix(k, GceLabelNameRolePrefix) {
+			continue
+		}
+		role := strings.TrimPrefix(k, GceLabelNameRolePrefix)
+		if role == "master" || role == "control-plane" {
+			isControlPlane = true
+		} else {
+			cm.Roles = append(cm.Roles, role)
+			cm.PrivateIP = fi.ValueOf(&instance.NetworkInterfaces[0].NetworkIP)
+			cm.ExternalIP = fi.ValueOf(&instance.NetworkInterfaces[0].AccessConfigs[0].NatIP)
+		}
+	}
+	if isControlPlane {
+		cm.Roles = append(cm.Roles, "control-plane")
+		cm.PrivateIP = fi.ValueOf(&instance.NetworkInterfaces[0].NetworkIP)
+		cm.ExternalIP = fi.ValueOf(&instance.NetworkInterfaces[0].AccessConfigs[0].NatIP)
+	}
```

Comment:
Why does this avoid setting the PrivateIP and ExternalIP when there is no role label? Why not unconditionally set those regardless of the role label?

Reply:
It doesn't, look at 282 to 284.